### PR TITLE
Nodes: Rework ShaderNodeElements and make some other refactorings

### LINKED ISSUE #23666: Convert remaining nodes and examples that are using the Nodes classes directly to using the ShaderNode functions
**Is your feature request related to a problem? Please describe.**

Currently some nodes and examples use the Nodes classes directly, while others use them through the ShaderNode functions. The ShaderNode approach is much simpler and easier to use than the direct - e.g. compare `new OperatorNode('*', new MathNode(MathNode.SIN, value1), new MathNode(MathNode.COS, value2))` and `mul(sin(value1), cos(value2))`.

**Describe the solution you'd like**

I suggest that the remaining nodes and examples that are using the Nodes classes directly will be converted to using the ShaderNode functions.
This will require https://github.com/mrdoob/three.js/issues/23564.

### LINKED ISSUE #25240: Circular dependency in ShaderNode
### Description

```
ShaderNode.js:3 Uncaught (in promise) ReferenceError: Cannot access 'ShaderNode' before initialization
    at Object.ShaderNode (ShaderNode.js:3:59)
    at eval (SkinningNode.js:8:87)
    at ./node_modules/three/examples/jsm/nodes/accessors/SkinningNode.js (chunk.vendors-node_modules_blueimp-md5_js_md5_js-node_modules_element-resize-detector_src_element-r-f0dec0.bfeca9936d49aeae8903.js:2501:1)
    at __webpack_require__ (chunk.app.0c10a065d695b4093c2e.js:107:42)
    at eval (ShaderNodeElements.js:248:85)
    at ./node_modules/three/examples/jsm/nodes/shadernode/ShaderNodeElements.js (chunk.vendors-node_modules_blueimp-md5_js_md5_js-node_modules_element-resize-detector_src_element-r-f0dec0.bfeca9936d49aeae8903.js:3590:1)
    at __webpack_require__ (chunk.app.0c10a065d695b4093c2e.js:107:42)
    at eval (ShaderNode.js:20:80)
    at ./node_modules/three/examples/jsm/nodes/shadernode/ShaderNode.js (chunk.vendors-node_modules_blueimp-md5_js_md5_js-node_modules_element-resize-detector_src_element-r-f0dec0.bfeca9936d49aeae8903.js:3568:1)
    at __webpack_require__ (chunk.app.0c10a065d695b4093c2e.js:107:42)
```

### Reproduction steps

use NodeMaterial and build with webpack

### Code

```js
// code goes here
```

### Live example

n/a

### Screenshots

_No response_

### Version

158

### Device

Desktop

### Browser

Chrome

### OS

Windows

### PR DESCRIPTION
**Pull request by LeviPesin**
Fixes https://github.com/mrdoob/three.js/issues/25240, partially fixes https://github.com/mrdoob/three.js/issues/23666
Implements https://github.com/mrdoob/three.js/issues/25240#issuecomment-1417067122 and https://github.com/mrdoob/three.js/issues/25240#issuecomment-1418100705

**Description**

Changes:
- Remove ShaderNodeElements and ShaderNodeBaseElements -- now the corresponding functions are exported by nodes files themself. Nothing changes for importing from `three/nodes`,
- Now some nodes assume that all nodes passed to them are in nodeObject form (user should manage that, but it's mostly managed by the nodes system),
- Introduce `addNodeElement`/`addLightNode`/`addNodeClass`/`addNodeMaterial` functions,
- Some refactoring introducing more of node chaining (it especially greatly simplifies functions) in most nodes files (examples can be refactored in a separate PR),
- Rename VarNode.add/sub/mul/div() to ShaderNode.addAssign()/subAssign()/etc (with semantics changed very slightly).

Tested all examples -- they seem to work.

### CODE REVIEW

**Comment by LeviPesin** [comment]
/ping @sunag @epreston This PR should resolve https://github.com/mrdoob/three.js/issues/25240.

**Comment by sunag** [comment]
@LeviPesin Awesome! It just broke `webgpu_lights_custom` example. I'm investigating...

File: examples/webgpu_lights_custom.html [hunk 1]
```diff
@@ -104,7 +104,7 @@
 
 					const { lightColor, reflectedLight } = inputs;
 
-					reflectedLight.directDiffuse.add( lightColor );
+					reflectedLight.directDiffuse.addAssign( lightColor );
```
Comment (LeviPesin) [inline examples/webgpu_lights_custom.html:107]: Missed that line, thank you!

**Comment by sunag** [comment]
Seems that we have just one last circular dependency in `NodeLoader.js`

```
npx madge examples/jsm/nodes/Nodes.js --circular --warning
```

image

**Comment by LeviPesin** [comment]
I think we can move nodeLib and fromType to a separate file and this should resolve the issue... Will do that.

**Comment by LeviPesin** [comment]
Done!

**Comment by sunag** [comment]
@LeviPesin We should give the option to expand the serialization, `Nodes.fromType()` is a good API, I think we should use a `nodeLib = new WeakMap` inside Nodes.js add method like `addNodeClass( SomeNode )` and `addNodeElement()`.

**Comment by LeviPesin** [comment]
Agreed! And same for materials.
Will do that soon...

**Comment by LeviPesin** [comment]
Done!
Remaining:
- What are the best names for `addNodeElement`/`addLightNode`/`addNodeClass`/`addNodeMaterial`? Maybe `register`, not `add`?
- Shouldn't we move `createNodeFromType` and `createMaterialFromType` to corresponding classes so that they would be static methods?
- Shouldn't we also move `addNodeElement`/`addLightNode`/`addNodeClass`/`addNodeMaterial` to classes?

**Comment by sunag** [comment]
@mrdoob do you have any suggestion about this? I'm thinking of merge and we deciding these definitions soon.

**Comment by sunag** [comment]
Modifications are more internal, so not affect the external API (examples/projects) or very little. Merging...

**Pull request merged**